association_backend/association/domain/services/update_draft_subscription.py

```diff
@@ -0,0 +1,31 @@
+import logging
+
+from association.domain.exceptions import SubscriptionNotUpdated
+from association.domain.repositories.association_repository import AssociationRepository
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+class SubscriptionUpdateInput(BaseModel):
+    session_id: str
+    customer_id: str
+    subscription_id: str
+
+
+async def update_draft_subscription(
+    data: SubscriptionUpdateInput, association_repository: AssociationRepository
+):
```

Comment:
let's avoid the term "draft", I feel like draft doesn't sound correct here, as it's not a document or something you are writing

We are in the process of creating a Subscription, so let's use PENDING to define the moment where you created it and are about to pay for it the first time